### PR TITLE
Added support for grabbing and putting data on S3.

### PR DESCRIPTION
**Pull request by shariqiqbal2810**
* Added S3DataGrabber to nipype/interfaces/io.py
* Added S3DataSink to nipype/interfaces/io.py
* Added tests to nipype/interfaces/tests/test_io.py
    * Currently the S3 tests are using a bucket on our lab's account.
    * We would like to have a conversation about a more long-term testing solution.

### CODE REVIEW

**Comment by satra** [comment]
for testing the grabber you can use the openfmri bucket (which is a public bucket). for the sink, i need to think of a better solution.

**Comment by shariqiqbal2810** [comment]
I changed the s3datagrabber test to use the openfmri data.  Unfortunately, these files are fairly large, so the tests are slowed (even though I'm only downloading 2).  Additionally, these modules require the use of boto3 (AWS Python API), and I was wondering what your solution would be to keep this optional. Currently I have this with my imports
```
try:
    import boto3
except:
    pass
```

**Comment by shariqiqbal2810** [comment]
FYI the only tests that are failing are due to a lack of boto3 on the testing system.  Tests all pass on my system.

**Comment by chrisfilo** [comment]
We should skip those tests on systems without boto3 (since not all users
will use it), but also add it to circle.yml config file so it would be
installed on the test server.

**Comment by shariqiqbal2810** [comment]
Ok done!  Any more thoughts on how to implement the S3DataSink test?

File: nipype/interfaces/tests/test_io.py [hunk 1]
```diff
@@ -8,10 +8,17 @@
...
+from nipype.testing import assert_equal, assert_true, assert_false, skipif
 import nipype.interfaces.io as nio
 from nipype.interfaces.base import Undefined
 
+noboto = False
+try:
+    import boto3
+except:
+    noboto = True
+    pass
```
Comment (chrisfilo) [inline nipype/interfaces/tests/test_io.py]: Codacy Issue found: [Unnecessary pass statement](https://www.codacy.com/app/krzysztof-gorgolewski/nipy_nipype/file/935437085/issues/source?bid=2767207&fileBranchId=2841000#l20)

**Comment by shariqiqbal2810** [comment]
So I think the only remaining things to sort out are:
* Figuring out a more permanent testing solution for S3DataSink
* Getting AWS credentials for the testing server

**Comment by chrisfilo** [comment]
What about using a fake S3 server? https://github.com/jubos/fake-s3

**Comment by shariqiqbal2810** [comment]
I modified the code to use a fake s3 server on the S3DataSink tests.  I added a 'gem install fakes3' line to the circle.yml file. Hopefully that will be enough to get everything working.

**Pull request merged**